pkg/resources/function_commons.go

```diff
@@ -0,0 +1,344 @@
+	"imports": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: what do we do with the version "latest".
+	"packages": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	"handler": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: use suppress from network policies when adding logic
+	"external_access_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		},
+		Optional:    true,
+		ForceNew:    true,
+		Description: "The names of [external access integrations](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) needed in order for this function’s handler code to access external networks. An external access integration specifies [network rules](https://docs.snowflake.com/en/sql-reference/sql/create-network-rule) and [secrets](https://docs.snowflake.com/en/sql-reference/sql/create-secret) that specify external locations and credentials (if any) allowed for use by handler code when making requests of an external network, such as an external REST API.",
+	},
+	"secrets": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"secret_variable_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The variable that will be used in handler code when retrieving information from the secret.",
+				},
+				"secret_id": {
+					Type:             schema.TypeString,
+					Required:         true,
+					Description:      "Fully qualified name of the allowed secret. You will receive an error if you specify a SECRETS value whose secret isn’t also included in an integration specified by the EXTERNAL_ACCESS_INTEGRATIONS parameter.",
+					DiffSuppressFunc: suppressIdentifierQuoting,
+				},
+			},
+		},
+		Description: "Assigns the names of secrets to variables so that you can use the variables to reference the secrets when retrieving information from secrets in handler code. Secrets you specify here must be allowed by the [external access integration](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) specified as a value of this CREATE FUNCTION command’s EXTERNAL_ACCESS_INTEGRATIONS parameter.",
+	},
+	// TODO [SNOW-1348103]: because of https://docs.snowflake.com/en/sql-reference/sql/create-function#id6, maybe it will be better to split into stage + path
+	"target_path": {
+		Type:     schema.TypeString,
+		Optional: true,
+		ForceNew: true,
+	},
+	"function_definition": {
+		Type:             schema.TypeString,
+		Required:         true,
+		ForceNew:         true,
+		DiffSuppressFunc: DiffSuppressStatement,
```

Comment:
Nit: wrap description with diffSuppressStatementFieldDescription

Reply:
Added in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3263